Lib/unittest/case.py

```diff
@@ -84,6 +84,27 @@ def testPartExecutor(self, test_case, isTest=False):
 def _id(obj):
     return obj
 
+
+_module_cleanups = []
+
+def addModuleCleanup(function, *args, **kwargs):
+    """Same as addCleanup, except the cleanup items are called even if
+    setUpModule fails (unlike tearDownModule)."""
+    _module_cleanups.append((function, args, kwargs))
+
+def doModuleCleanups():
+    """Execute all module cleanup functions. Normally called for you after
+    tearDownModule."""
+    exceptions = []
+    while _module_cleanups:
+        function, args, kwargs = _module_cleanups.pop()
+        try:
+            function(*args, **kwargs)
+        except Exception as exc:
+            exceptions.append(exc)
+    if exceptions:
+        raise exceptions[0]
```

Comment:
I totally get why you have to do it this way, although it would be nice if we had trio's [MultiError](https://trio.readthedocs.io/en/latest/reference-core.html#trio.MultiError) in the stdlib. :)  It's unfortunate that we have to throw away all the other exceptions, but I don't know where you can stash them.  E.g. IIUC, instance cleanups stash the exceptions on `self.errors`, but there's no such place on modules (unless this code adds it, which, yuck :)).  The other alternative is to create something like a simple `MultiError`, add the exceptions to that, and then raise the `MultiError` (maybe only if `len(exceptions) > 1`?).

It seems like a lot of extra complication for hopefully an, um, exceptional case, so maybe punt on that for now.  IOW, I'm cool with this.